core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1037,21 +1037,52 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * Convert a passed byte string (e.g. 50b, 100kb, or 250mb) to bytes for
+   * internal use.
+   *
+   * If no suffix is provided, the passed number is assumed to be in bytes.
+   */
+  def byteStringAsBytes(str: String): Long = {
+    JavaUtils.byteStringAsBytes(str)
+  }
+
+  /**
+   * Convert a passed byte string (e.g. 50b, 100kb, or 250mb) to kilobytes for
```

Comment:
Tough call. I like correctness; I do think people aren't used to seeing `KiB`. They will probably figure it out quickly. I suppose it would help if later we needed to add support for real kilobytes somewhere, since `KB` isn't taken.